1st-gen/tools/truncated/README.md

```diff
@@ -1,49 +1,241 @@
+
+```html demo
+<div
+    style="width: 150px; border: 1px solid var(--spectrum-gray-300); padding: 8px;"
+>
+    <sp-truncated>
+        ThisIsAVeryLongWordWithoutAnySpacesThatShouldBeTruncated
+    </sp-truncated>
+</div>
+```
+
+#### Custom overflow content
+
+By default, tooltip text will be extracted from the overflowing content. To provide your own custom overflow content for the tooltip, slot it into the `overflow` slot:
+
+```html demo
+<div
+    style="width: 200px; border: 1px solid var(--spectrum-gray-300); padding: 8px;"
+>
+    <sp-truncated>
+        This is the inline content that gets truncated
+        <span slot="overflow">
+            This custom overflow content will appear in the tooltip with any
+            additional information or formatting you need.
+        </span>
+    </sp-truncated>
+</div>
+```
+
+#### Tooltip placement
+
+You can control the placement of the tooltip using the `placement` attribute. The default placement is `top-start`.
+
+```html demo
+<div style="display: flex; gap: 16px; flex-wrap: wrap;">
+    <div
+        style="width: 150px; border: 1px solid var(--spectrum-gray-300); padding: 8px;"
+    >
+        <sp-truncated placement="top">
+            Top placement for this truncated text
+        </sp-truncated>
+    </div>
+    <div
+        style="width: 150px; border: 1px solid var(--spectrum-gray-300); padding: 8px;"
+    >
+        <sp-truncated placement="bottom">
+            Bottom placement for this truncated text
+        </sp-truncated>
+    </div>
+    <div
+        style="width: 150px; border: 1px solid var(--spectrum-gray-300); padding: 8px;"
+    >
+        <sp-truncated placement="left">
+            Left placement for this truncated text
+        </sp-truncated>
+    </div>
+    <div
+        style="width: 150px; border: 1px solid var(--spectrum-gray-300); padding: 8px;"
+    >
+        <sp-truncated placement="right">
+            Right placement for this truncated text
+        </sp-truncated>
+    </div>
+</div>
+```
+
+#### Custom success message
+
+When text is copied to the clipboard, a custom success message can be displayed:
+
+```html demo
+<div
+    style="width: 200px; border: 1px solid var(--spectrum-gray-300); padding: 8px;"
+>
+    <sp-truncated success-message="Email copied successfully!">
+        iamaverylongusernametoshowthecustom.message.text@example.com
+    </sp-truncated>
+</div>
+```
+
+### Usage in patterns
 
-By default, tooltip text will be extracted from overflowing content. To provide your own overflow content, slot it into "overflow":
+#### With field labels
+
+`<sp-truncated>` is commonly used with form fields to display long values that might overflow:
+
+```html demo
+<sp-field-label for="email-field">Email address</sp-field-label>
+<div
+    style="width: 250px; border: 1px solid var(--spectrum-gray-300); padding: 8px; border-radius: 4px;"
+>
+    <sp-truncated>very.long.email.address@subdomain.example.com</sp-truncated>
+</div>
```

Comment:
Should this be a field instead of a div? Field labels should only be used with interactive elements, such as form fields.